### ISSUE TITLE
Nested calls to rc.context() fail

### ISSUE THREAD

**Issue by lukelbd** [lukelbd/proplot#80]
You can end up with persistent changes to the `rc` object if `rc.context()` statements are nested. Most notably

```python
import proplot as plot
with plot.rc.context(...):
    f, ax = plot.subplots()
    ax.format(...)
```

fails because `ax.format` calls `rc.context(...)` *internally*. #50 will fix this (I ended up putting some quite tangentially related stuff in that PR... will try to avoid this in the future).

**Comment by lukelbd** [comment]
Actually `rc.context` is completely broken... because `Axes.__init__()` calls `self.format(mode=1)` to implement the custom proplot `rcParamsLong` params (e.g. `abc`, `abc.style`, `land`, `land.color`, etc.).

**Comment by lukelbd** [comment]
Instead of waiting for #50 to be merged, I'll try to copy the changes in #50 unrelated to `add_subplot`, `GeometrySolver`, and `EdgeStack` into the main branch. That will resolve this issue.

I originally put so much extra stuff in #50 because I thought it would be finished in November; now it looks like it will have to wait until the middle of next year sometime. Learned a hard lesson about dividing up tasks into small PRs.

**Comment by lukelbd** [comment]
Closed by #91
